src/include/btmem.h

```diff
@@ -997,7 +997,7 @@ struct __wt_update {
 						   finalized prepare */
 #define	WT_UPDATE_STATE_LOCKED		1	/* locked */
 #define	WT_UPDATE_STATE_PREPARED	2	/* prepared */
-	uint8_t state;			/* state (one byte : conserve memory) */
+	volatile uint8_t state;
```

Comment:
I don't see how this change fits in - why is it required?

Reply:
This was done in response to a comment I made, and Keith said something similar (somewhere I can't find right now).

That is, the locking protocol requires that `state` is switched to `WT_UPDATE_STATE_LOCKED` before overwriting the timestamp as part of committing a prepared transaction.  Readers must see the `state` change so they can be sure they have a valid timestamp.  Making the field volatile is necessary but not sufficient on CPUs with weak memory models.